### ISSUE TITLE
[12.0][l10n_it_fiscalcode] Codice fiscale per la persona dovrebbe avere caratteri in maiuscolo

### ISSUE THREAD

**Issue by c-amara** [OCA/l10n-italy#2250]
Versioni coinvolte:

- [ ] 12.0

Passi per riprodurre:
Creare o modificare un contatto "persona"

Comportamento osservato:
Se scritto con caratteri minuscoli rimane tale. E inoltre la creazione di una fattura elettronica XML fallisce se il codice fiscale viene scritto in minuscolo

Comportamento atteso:
Il codice fiscale di una persona per convenzione è sempre riportato in maiuscolo. 

Video/Screenshot (opzionale):

**Comment by c-amara** [comment]
Provo ad inviare una PR. Ditemi se siete d'accordo

**Comment by SimoRubi** [comment]
In v10 il codice fiscale viene portato a maiuscolo da https://github.com/OCA/l10n-italy/pull/607 quando https://github.com/OCA/web/tree/10.0/web_widget_char_switchcase è installato.
Potresti portare a v12 quelle modifiche, dopo la migrazione a v12 del modulo https://github.com/OCA/web/tree/10.0/web_widget_char_switchcase.

Non sapevo fosse un requisito per la generazione della fattura elettronica, potresti riportare l'errore che viene sollevato?

**Comment by sergiocorato** [comment]
Concordo, non credo sia obbligatorio ma vederlo in minuscolo fa osceno in qualsiasi nazione...
Direi che senza scomodare moduli aggiuntivi si potrebbe mettere una funzione direttamente.

**Comment by c-amara** [comment]
> 
> 
> In v10 il codice fiscale viene portato a maiuscolo da #607 quando https://github.com/OCA/web/tree/10.0/web_widget_char_switchcase è installato.
> Potresti portare a v12 quelle modifiche, dopo la migrazione a v12 del modulo https://github.com/OCA/web/tree/10.0/web_widget_char_switchcase.
> 
> Non sapevo fosse un requisito per la generazione della fattura elettronica, potresti riportare l'errore che viene sollevato?

L'errore è il seguente:

Type {http://ivaservizi.agenziaentrate.gov.it/docs/xsd/fatture/v1.2}CodiceFiscaleType pattern constraint violated by value "codicefiscaleminuscolo"

Per quanto riguarda `web_widget_char_switchcase` mi sembra che non sia disponibile la versione per la 12. Per questo ho optato per questa soluzione anche veloce.

**Comment by c-amara** [comment]
Ovviamente Cambiando in maiuscolo il codice fiscale tutto fila liscio

**Comment by sergiocorato** [comment]
Non vedo la PR

**Comment by c-amara** [comment]
Eccola https://github.com/OCA/l10n-italy/pull/2253 scusami! Non l'avevo più creata

**Comment by TheMule71** [comment]
Aggiunto #2371
